README.md

```diff
@@ -1,52 +1,167 @@
 
-Instead of relying on a gem or library to do the heavy lifting for us, we can use Ajax to do it ourselves, without having a page refresh. 
+** Maybe link this back to rails or a gem example **
 
+As you can see, getting data from an API is pretty easy but we haven't really said what an API is.
 
-## API Basics
+```ruby
+github_commits = Github::Client::Repos::Commits.new
+github_commits.all(user: 'rails', repo: 'rails')
+github_commits.all(user: 'rails', repo: 'rails', author: 'DHH')
+```
 
-Underneath the hood of all those gems, a request as been made to the API (to a specific url) and then the app responds to that request with the data.
+## Defining API (done)
 
-use postman chrome extension or app?? to make API request to demonstrate getting back JSON and manipulating it (screenshots) from the Github API
+For the purpose of this lesson, we are mostly concerned with web APIs.  But the term API actually has a more broad meaning.
 
-theoretical definition of API
+> In computer programming, an application programming interface (API) is a set of routines, protocols, and tools for building software and applications. - Wikipedia
+
+In its simplest form, an API in relation to Object Oriented programming is a class and the list of methods we define.  When creating a class, we are defining a guidebook on how to interact with other parts of the code.  We get to decide which methods and variables are public or private, essentially controlling how to interact with the class.  When we apply this concept to the web, we get web APIs like the GitHub API and Twitter API. From our Postman experiment, we saw how GitHub provides a way for us interact with the data on their system. Just like how a class provides a set of public methods to interact with, web APIs provide us with urls. The list of urls that GitHub provides on https://developer.github.com/v3 act as the public methods into their system.  The developers that created the API control which resources they want to share and who has access to them. 
 
 ## Ajax
 
-describe what ajax is and what it's used for
+*** SOMETHING HERE to make it flow better ***
+
+Asynchrounous Javascript and XML (Ajax) is a technique that is used in web applications.  It provides a way for content to be retrieved from a server and displayed without having to refresh the entire page.  Remember in your **blank application** where you had to create a list of **objects**?  Each time we filled out the form and clicked on **create**, the web browser would have to refresh the entire page to see the results sent back from our server.
+
+Single page applications provide a better user experience by allowing users to manipulate data on the server and see the results without having to refresh the page.  This is the power of Ajax in action.  In the background, requests are made to a web API using JavaScript.  As developers we can then choose to alter the displayed HTML based on the responses from the web API.
```

Comment:
It’s important to note that AJAX has benefits outside of a single-page applications (and that SPAs aren’t all unicorns and rainbows). A little bit of AJAX on an otherwise static site (say, for clicking a button or submitting a form) can feel magical.